Tasks/AzureResourceManagerTemplateDeploymentV3/models/ARMDeployTaskParameters.ts

```diff
@@ -0,0 +1,112 @@
+    public endpointPortalUrl: string;
+    public deploymentName: string;
+    public deploymentMode: string;
+    public credentials: msRestAzure.ApplicationTokenCredentials;
+    public deploymentOutputs: string;
+    public addSpnToEnvironment: boolean;
+    public connectedService: string;
+    public deploymentScope: string;
+    public managementGroupId: string;
+
+    private async getARMCredentials(connectedService: string): Promise<msRestAzure.ApplicationTokenCredentials> {
+        var azureEndpoint = await new AzureRMEndpoint(connectedService).getEndpoint();
+        return azureEndpoint.applicationTokenCredentials;
+    }
+
+    public async getARMTemplateDeploymentTaskParameters() : Promise<ARMDeployTaskParameters>
+    {
+        try {
+
+            //Deployment Scope
+            this.deploymentScope = tl.getInput("deploymentScope");
+            if(!this.deploymentScope){
+                this.deploymentScope = "Resource Group";
+            }
+
+            var resourceGroupNameInServiceConnection;
+
+            //Service Connection
+            this.connectedService = tl.getInput("ConnectedServiceName", true);
+            var endpointTelemetry = '{"endpointId":"' + this.connectedService + '"}';
+            console.log("##vso[telemetry.publish area=TaskEndpointId;feature=AzureResourceManagerTemplateDeployment]" + endpointTelemetry);
+            this.endpointPortalUrl = tl.getEndpointDataParameter(this.connectedService, "armManagementPortalUrl", true);
+            var armServiceConnectionScope = tl.getEndpointDataParameter(this.connectedService, 'ScopeLevel', true);
+            if(!!armServiceConnectionScope && armServiceConnectionScope === "Subscription"){
+                var armServiceConnectionAuthScope = tl.getEndpointAuthorizationParameter(this.connectedService, 'scope', true);
+                if(!!armServiceConnectionAuthScope){
+                    var armServiceConnectionAuthScopeSplit = armServiceConnectionAuthScope.split("/");
+                    if(!!armServiceConnectionAuthScopeSplit[4]){
+                        armServiceConnectionScope = "Resource Group";
+                        resourceGroupNameInServiceConnection = armServiceConnectionAuthScopeSplit[4];
+                    }
+                }
+            }
+            console.log(tl.loc("ARMServiceConnectionScope", armServiceConnectionScope));
+
+            //Management Group Id
+            if(this.deploymentScope === "Management Group"){
+                this.managementGroupId = tl.getEndpointDataParameter(this.connectedService, "ManagementGroupId", false);
+            }
+
+            //Subscripion Id
+            this.subscriptionId = tl.getInput("subscriptionName");
+            if(!this.subscriptionId && (this.deploymentScope === "Subscription" || this.deploymentScope === "Resource Group")) {
+                this.subscriptionId = tl.getEndpointDataParameter(this.connectedService, "SubscriptionId", false);
```

Comment:
what is the error message which `getEndpointDataParameter` throws if parameter does not exist? Is it intuitive for use? If no, we should catch here and throw proper error message.

Reply:
Error: Endpoint data parameter <endpointId> not present: <parameter name>
e.g. 
Error: Endpoint data parameter d6a95b0b-96c5-4795-8916-11111d3b1111 not present: subscriptionId